### PR TITLE
Finding a maximum and minimum element of an array using the divide and conquer algorithm.

### LINKED ISSUE #4504: Finding  the minimum and maximum element in an array using Divide and Conquer
### Description

In this document , I will first explain what is Divide and Conquer algorithm in brief. Secondly, I will also provide the algorithm and code to find  the minimum and maximum element in an array using Divide and Conquer. 

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by am-dusky**
<hr>

## Description 📜

In this document I have explained   Divide and Conquer algorithm in brief. Secondly, I have  explained the  naive as well as  Divide and Conquer approach to find the minimum and maximum element in an array . I have also explained advantage and disadvantage of using Divide and Conquer approach.

I also want this PR to be a part of hacktoberfest , so pls add hacktoberfest lable to it.
Fixes #4504 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by ChetasShree** [comment]
Amazing Work @am-dusky !!
@Suvraneel @Saviour1001 pls review this PR and merge it :)
Thank you !!

File: C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+T( n ) = 2 T ( n/2 ) + 2
+
+After solving this recurrence relation we will get,
+
+T ( n ) = 3n / 2 - 2
+
+Therefore, the number of comparisons will be 3n / 2 - 2.
+
+#### Conclusion
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md:140]:
```suggestion
### Conclusion

```

File: C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+```
+Output
+```
+Enter the size of  array : 7
+Enter elements to the array : 23 56 11 -34 0 98 3
+Maximum element is: 98
+Minimum element is: -34
+```
+
+Analysis:
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md:121]:
```suggestion
#### Analysis:

```

File: C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    max_min(arr,0,n-1,min,max);
+    //displaying maximum and minimum elements
+    cout<<"Maximum element is: "<<max<<"\n";
+    cout<<"Minimum element is: "<<min;
+
+    return 0;
+}
+
+```
+Output
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md:113]:
```suggestion
#### Output

```

File: C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    int min=arr[0];
+    for (int i=1;i<n-1;i++){
+        if(arr[i]>max) //comparison (n-1) times
+            max=arr[i];//stores max element
+        if(arr[i]<min)//comparison (n-1) times
+             min=arr[i];//stores minimum element
+    }
+}
+```
+Analysis:
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md:37]:
```suggestion
#### Analysis:

```

File: C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+Finding a maximum and minimum element of an array
+
+solution:
+
+To find maximum and minimum numbers in a given array (arr[]) of size n, the following algorithm can be used.
+
+1. Naive approach
+2. Using the divide and conquer approach
+
+#### Naive method
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Divide & Conquer/max_and_min_using_dac.md:22]:
```suggestion
### Naive method

```

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Comment by Suvraneel** [comment]
@am-dusky  Getting pro at this 🚀 ❤️ :octocat:

**Pull request merged**